src/app/modules/bookmark/component/bookmark-settings/bookmark-settings.component.ts

```diff
@@ -29,6 +31,19 @@ export class BookmarkSettingsComponent implements UserSettingsComponent {
     this.addBookmark()
   }
 
+  public onAddFileClick(): void {
+    (this.input.nativeElement as HTMLInputElement).click();
+  }
+
+  public onFileSelected(event: InputEvent): void {
```

Comment:
Did you check if cancelling the file selection doesn't trigger/call this method? Otherwise you'd be adding a bookmark with an empty path reading just `file:///`

Reply:
yea hitting cancel doesn't trigger the `(change)` event so it doesn't do anything